osquery/events/linux/bpf/bpfeventpublisher.cpp

```diff
@@ -349,57 +302,52 @@ Status BPFEventPublisher::run() {
         "Halting the publisher since initialization has failed");
   }
 
-  ebpfpub::IPerfEventReader::ErrorCounters error_counters{};
-  auto last_error_counters_report = getUnixTime();
+  BPFErrorState bpf_error_state;
+
+  auto last_error_report = getUnixTime();
+  auto last_tracker_restart = getUnixTime();
 
   while (!isEnding()) {
+    auto current_time = getUnixTime();
+    if (last_tracker_restart + (FLAGS_bpf_state_tracker_reset_time * 60) <
+        current_time) {
```

Comment:
This is wall clock time, which means in practice it will reset every time a laptop awakes from sleep. Is that okay? (I'm not really sure I understand what this does)

Reply:
A procfs snapshot is taken when the publisher is first started; the event stream will then operate on that information to keep it up to date. Errors in the event capture (like lost events) will gradually deteriorate the quality of this snapshot. This PR introduces a reset once every `FLAGS_bpf_state_tracker_reset_time` minutes